python/cudf/cudf/_lib/aggregation.pyx

```diff
@@ -241,24 +243,18 @@ cdef class _AggregationFactory:
         cdef string cpp_str
 
         # Handling UDF type
-        nb_type = numpy_support.from_dtype(kwargs['dtype'])
+        nb_type = numpy_support.from_dtype(kwargs['dtype'].to_numpy)
         type_signature = (nb_type[:],)
         compiled_op = cudautils.compile_udf(op, type_signature)
-        output_np_dtype = np.dtype(compiled_op[1])
+        output_np_dtype = cudf_dtype(np.dtype(compiled_op[1]))
         cpp_str = compiled_op[0].encode('UTF-8')
-        if output_np_dtype not in np_to_cudf_types:
+        if cudf_dtype(output_np_dtype) not in np_to_cudf_types:
             raise TypeError(
                 "Result of window function has unsupported dtype {}"
                 .format(op[1])
             )
```

Comment:
It shouldn't need to check the dict, that's correct. The dtype should error upon construction. The only question is if this case deserves its own error. To figure that out I think I need to look closely here and figure out what exactly the user could be doing that would cause them to run into this.